### PR TITLE
Initial unit tests for pbench-run-benchmark

### PR DESCRIPTION
**Pull request by portante**
We have two separate commits here, one to fix some whitespace formatting issues, and the second which introduces a number of changes to the `pbench-run-benchmark` environment to run unit tests and correct implementation errors found as a result.

It is not a thorough set of unit tests, but it exercises a fair amount of the code base to start.

### CODE REVIEW

**Comment by portante** [comment]
Unfortunately, until we merge PR #2005, we have flaky `util-scripts` unit tests.

File: agent/bench-scripts/pbench-run-benchmark.pl [hunk 1]
```diff
@@ -36,19 +36,24 @@
...
+    printf "\t%s\n",  join(" ", @benchmarks);
+    exit 1;
 }
 my $benchmark = shift(@ARGV);
 if ($benchmark eq "list") {
-    my @benchmarks = get_benchmark_names(get_pbench_bench_config_dir);
     printf "%s\n",  join(" ", @benchmarks);
     exit;
 }
+my %benchmarks_hash = map { $_ => 1 } @benchmarks;
```
Comment (dbutenhof) [inline agent/bench-scripts/pbench-run-benchmark.pl:51]: Ah, Perl idioms. I do wish I could have not recovered this arcane trivia...

**Review by dbutenhof — APPROVED** [review]
The rust on my perl is getting rusty, but I think this makes sense. ;-)

**Comment by portante** [comment]
This is now based on top of PR #2071 to avoid conflicts in the test names, and that PR is more important to land than this one.

**Review by ndokos — CHANGES_REQUESTED** [review]

**Review by dbutenhof — APPROVED** [review]

**Review by portante — COMMENTED** [review]
Rebased, ready for review.

**Review by dbutenhof — APPROVED** [review]

File: agent/bench-scripts/pbench-run-benchmark.pl (hunk 1 above)
Comment (ndokos) [reply agent/bench-scripts/pbench-run-benchmark.pl:51]: The sad thing is that I figured it out during the last review and now I cannot remember it any more...

**Review by ndokos — APPROVED** [review]

**Comment by portante** [comment]
Adjusted commit on `main` at: https://github.com/distributed-system-analysis/pbench/commit/6673fe2209fbfc22a20a2d912e7db468e3fce595.

**Pull request merged**